app/components/container/containerController.js

```diff
@@ -179,21 +126,29 @@ function ($q, $scope, $state, $transition$, $filter, Container, ContainerCommit,
     $scope.container.edit = false;
   };
 
-  $scope.containerLeaveNetwork = function containerLeaveNetwork(container, networkId) {
-    $('#loadingViewSpinner').show();
-    Network.disconnect({id: networkId}, { Container: $transition$.params().id, Force: false }, function (d) {
+  function containerNetworkAction(container, networkId, Spin, Action, Obj) {
+    $('#'+Spin+'Spinner').show();
+    Network[Action[0]]({id: networkId}, Obj, function (d) {
       if (container.message) {
-        $('#loadingViewSpinner').hide();
-        Notifications.error('Error', d, 'Unable to disconnect container from network');
+        $('#'+Spin+'Spinner').hide();
+        Notifications.error('Error', d, 'Unable to '+Action[0]+' container '+Action[1]+' network');
       } else {
-        $('#loadingViewSpinner').hide();
-        Notifications.success('Container left network', $transition$.params().id);
+        $('#'+Spin+'Spinner').hide();
+        Notifications.success('Container '+Action[2]+' network', $transition$.params().id);
         $state.go('container', {id: $transition$.params().id}, {reload: true});
       }
     }, function (e) {
-      $('#loadingViewSpinner').hide();
-      Notifications.error('Failure', e, 'Unable to disconnect container from network');
+      $('#'+Spin+'Spinner').hide();
+      Notifications.error('Failure', e, 'Unable to '+Action[0]+' container '+Action[1]+' network');
     });
```

Comment:
We can simplify this function by moving `hide` function to the `finally` handler.

Reply:
I am not sure about it. `.finally` is already used in other functions (e.g., `$scope.remove`), but I don't know how to use it in `containerNetworkAction`. More precisely, `ContainerService.remove` takes two arguments and `Network[Action[0]]` takes three.

Reply:
Well, at least for `then` branch we could hide spinner regardless of selected branch